### ISSUE TITLE
Project Ozone 3 support (1.12.2)

### ISSUE THREAD

**Issue by versedwildcat** [Radk6/MC-Optimization-Guide#7]
Heyo, I tried applying the same steps as in most other cases to project ozone 3.

Installed Cleanroom Loader on PrismLauncher, migrated .minecraft folder from previous instance to it. 

Removed:

diethopper-1.1.jar.disabled
FastWorkbench-1.12.2-1.7.4.jar.disabled
fencejumper-1.12-1.0.5.jar.disabled
foamfix-0.10.15-1.12.2.jar.disabled
Forgelin-1.8.4.jar.disabled
jei_1.12.2-4.16.1.302.jar.disabled
Mekanism-1.12.2-9.8.3.390.jar.disabled
MekanismGenerators-1.12.2-9.8.3.390.jar.disabled
MekanismTools-1.12.2-9.8.3.390.jar.disabled
NotEnoughIDs-1.5.4.4.jar.disabled
phosphor-forge-mc1.12.2-0.2.7-universal.jar.disabled
Toast Control-1.12.2-1.8.1.jar.disabled
AttributeFix-1.12.2-1.0.10.jar.disabled
BetterFps-1.4.8.jar.disabled

Added:

StellarCore-1.5.13.jar
Alfheim-1.4.jar
betterbiomeblend-1.12.2-1.1.7-forge.jar
Forgelin-Continuous-2.0.21.0.jar
+Fugue-1.12.2-0.16.4.jar
HadEnoughItems_1.12.2-4.26.1.jar
RoughlyEnoughIDs-2.2.1.jar
Mekanism-Community-Edition-1.12.2-9.12.10-Core.jar
Mekanism-Community-Edition-1.12.2-9.12.10-Generators.jar
Mekanism-Community-Edition-1.12.2-9.12.10-Tools.jar
OptiFine_1.12.2_HD_U_G5.jar
UniversalTweaks-1.12.2-1.13.0.jar
!Red-Core-MC-1.8-1.12-0.6-Dev-8.jar
RenderLib-1.12.2-1.3.5.jar
vintagefix-0.5.1.jar
censoredasm5.20.jar
EntityCulling-1.12.2-6.4.3.jar
Nothirium-1.12.2-0.3.4-beta.jar
scalar-1.12.2-2.11.1-2.jar

Also applied the necessary compatibility options for the various mods listed on https://github.com/Radk6/MC-Optimization-Guide/blob/main/mods-n-stuff/1.12.2.md

The game crashes before any window appears.
Here is the log of the latest crash:
[crash_log.txt](https://github.com/user-attachments/files/17748958/crash_log.txt)

Last few lines:
```[14:53:22] [Client thread/INFO] [Quark ASM]: Located Method, patching...
[14:53:22] [Client thread/INFO] [Quark ASM]: Patch result: true
[14:53:22] [Client thread/WARN] [Foundation]: org.objectweb.asm.tree.MethodNode can't handle Java 21 class, please port the mod (if you are the author)!
[14:53:22] [Client thread/INFO] [EnderCore ASM]: Transforming Class [net.minecraft.entity.ai.EntityAICreeperSwell], Method [func_75246_d]
[14:53:22] [Client thread/INFO] [EnderCore ASM]: Transforming net.minecraft.entity.ai.EntityAICreeperSwell Finished.
[14:53:22] [Client thread/INFO] [Apotheosis : Core]: Transforming EntityAITempt...
[14:53:22] [Client thread/INFO] [Apotheosis : Core]: Successfully transformed EntityAITempt
[14:53:22] [Client thread/WARN] [Foundation]: openmods.core.fixes.HorseNullFix$Llama can't handle Java 21 class, please port the mod (if you are the author)!
[14:53:22] [Client thread/INFO] [Quark ASM]: Transforming net.minecraft.item.crafting.RecipesBanners$RecipeAddPattern
[14:53:23] [Client thread/INFO] [Quark ASM]: Applying Transformation to method (Names [matches, func_77569_a] Descriptor (Lnet/minecraft/inventory/InventoryCrafting;Lnet/minecraft/world/World;)Z)
[14:53:23] [Client thread/INFO] [Quark ASM]: Located Method, patching...
[14:53:23] [Client thread/INFO] [Quark ASM]: Located patch target node INVOKESTATIC net/minecraft/tileentity/TileEntityBanner.func_175113_c (Lnet/minecraft/item/ItemStack;)I
[14:53:23] [Client thread/INFO] [Quark ASM]: Patch result: true
[14:53:23] [Client thread/FATAL] [Foundation]: Unable to launch
java.lang.reflect.InvocationTargetException: null
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at top.outlands.foundation.LaunchHandler.launch(LaunchHandler.java:119) [LaunchHandler.class:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at top.outlands.foundation.boot.Foundation.main(Foundation.java:41) [foundation-0.13.9.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:105) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?]
Caused by: java.lang.AbstractMethodError: Receiver class net.minecraft.nbt.NBTTagByte does not define or inherit an implementation of the resolved method 'abstract void stellar_core$setPooled(boolean)' of interface github.kasuminova.stellarcore.mixin.util.StellarPooledNBT.
	at github.kasuminova.stellarcore.common.pool.NBTTagPrimitivePool.lambda$static$6(NBTTagPrimitivePool.java:33) ~[NBTTagPrimitivePool.class:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024) ~[?:?]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762) ~[?:?]
	at github.kasuminova.stellarcore.common.pool.NBTTagPrimitivePool.<clinit>(NBTTagPrimitivePool.java:33) ~[NBTTagPrimitivePool.class:?]
	at net.minecraft.nbt.NBTTagCompound.setInteger(NBTTagCompound.java:656) ~[fy.class:?]
	at net.minecraft.item.ItemStack.writeToNBT(ItemStack.java:233) ~[aip.class:?]
	at net.minecraft.item.ItemStack.getTextComponent(ItemStack.java:1101) ~[aip.class:?]
	at net.minecraft.stats.StatList.initMiningStats(StatList.java:181) ~[qs.class:?]
	at net.minecraft.stats.StatList.init(StatList.java:125) ~[qs.class:?]
	at net.minecraft.init.Bootstrap.register(Bootstrap.java:515) ~[ni.class:?]
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:365) ~[bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) ~[Main.class:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	... 8 more
Process exited with code 1.
```

Any pointers or advice? My gut says it's possible given that PO3 shares 90% of its mods with the other packs on the list.
If this is the wrong place for this or you'd prefer to talk somewhere else let me know!
Thank you

**Comment by Radk6** [comment]
I haven't started troubleshooting the pack yet, but one thing I noticed is that you're using EntityCulling by Meldexun. I probably should've made it clearer, but the "intended" mod is Entity Culling by tr7zw.

**Comment by Radk6** [comment]
I added Project Ozone 3, but the issues aren't fixed yet. I'll look into it when I have more time.

**Comment by Radk6** [comment]
[I reported the CensoredASM issue](https://github.com/LoliKingdom/LoliASM/issues/263). 

As for the Dimensional Edibles: Nomifactory Edition one, I'm not sure. I initially thought it was caused by Nothirium and/or OptiFine, but in a completely clean instance with just those mods and their dependencies, it worked fine.

Anyways, these issues seem pretty minor and non game-breaking. I'll close this and add them to the "known issues" part under PO3.
